### PR TITLE
Add a curlrc file

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

This adds a curlrc file with sane defaults including retries etc. which can be used for all our Docker builds without having to specify all flags (e.g. fail) every time

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: stackable-base/Dockerfile [hunk 1]
```diff
@@ -65,6 +65,8 @@ SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
 # https://github.com/stackabletech/docker-images/pull/533
 COPY stackable-base/stackable/dnf.conf /etc/dnf/dnf.conf
 
+# Default curl configuration to avoid forgetting settings and to declutter the Dockerfiles
+COPY stackable-base/stackable/curlrc /root/.curlrc
```
Comment (NickLarsenNZ) [inline stackable-base/Dockerfile:69]: Will we also need it for whatever user we switch to (I haven't checked if we do switch users and then curl stuff)?

File: stackable-base/Dockerfile [hunk 2]
```diff
@@ -140,6 +142,9 @@ echo -e "if [ -f ~/.bashrc ]; then\n\tsource ~/.bashrc\nfi" >> /stackable/.profi
 chown stackable:stackable /stackable/.bashrc
 chown stackable:stackable /stackable/.profile
 
+cp /root/.curlrc /stackable/.curlrc
+chown stackable:0 /stackable/.curlrc
```
Comment (NickLarsenNZ) [inline stackable-base/Dockerfile:146]: Nice, was going to mention doing it for any users we switch to (so deleting my previous comment).

File: tools/Dockerfile [hunk 3]
```diff
@@ -34,10 +34,10 @@ WORKDIR /stackable/bin
 ENV PATH=/stackable/bin:$PATH
 
 # Get latest stable version from curl -L -s https://dl.k8s.io/release/stable.txt
-RUN	curl -L https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl \
+RUN	curl https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl \
```
Comment (NickLarsenNZ) [inline tools/Dockerfile:37]: Replace tab with space

```suggestion
RUN curl https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl \
```

File: tools/Dockerfile [hunk 4]
```diff
@@ -34,10 +34,10 @@ WORKDIR /stackable/bin
...
     -o /stackable/bin/kubectl && chmod +x /stackable/bin/kubectl
 
-RUN	curl -L https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64 \
+RUN	curl https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64 \
```
Comment (NickLarsenNZ) [inline tools/Dockerfile:40]: Replace tab with space

```suggestion
RUN curl https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64 \
```

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM

File: stackable-base/Dockerfile (hunk 1 above)
Comment (NickLarsenNZ) [reply stackable-base/Dockerfile:69]: As per future comment, this was already considered and done.

File: tools/Dockerfile (hunk 3 above)
Comment (lfrancke) [reply tools/Dockerfile:37]: THERE'S A TAB IN THERE?

**Comment by lfrancke** [comment]
I'm running a few tests before "officially" requesting a review but thank you very much for jumping on it!

File: tools/Dockerfile (hunk 3 above)
Comment (NickLarsenNZ) [reply tools/Dockerfile]: Do you have your tab size set to one space? I have mine set to 4 so it is obvious, not that I viewed this change locally)

**Review by NickLarsenNZ — APPROVED** [review]
LGTM, just some lint errors:

```
testing-tools/Dockerfile:26 DL3008 warning: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`
testing-tools/Dockerfile:61 DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`
airflow/Dockerfile:44 DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`
```

I'm unsure what we want to do with those. Pin the apt packages, or set an ignore rule?

**Comment by lfrancke** [comment]
I don't know...it's been something I've been meaning to look at for ages but didn't get around to yet.

**Pull request merged**